extension/src/popup/views/AddAccount/AddAccount/index.tsx

```diff
@@ -35,30 +36,60 @@ export const AddAccount = () => {
   const { t } = useTranslation();
   const dispatch: AppDispatch = useDispatch();
   const authError = useSelector(authErrorSelector);
+  const hasPrivateKey = useSelector(hasPrivateKeySelector);
 
-  const handleSubmit = async (values: FormValues) => {
-    const { password } = values;
+  // In case a password is not provided here popupMessageListener/addAccount
+  // will try to use the existing password value saved in the session store
+  const handleAddAccount = useCallback(
+    async (password: string = "") => {
+      const res = await dispatch(addAccount(password));
 
-    const res = await dispatch(addAccount(password));
+      if (addAccount.fulfilled.match(res)) {
+        emitMetric(METRIC_NAMES.accountScreenAddAccount, {
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          number_of_accounts: res.payload.allAccounts.length,
+        });
+        navigateTo(ROUTES.account);
+      }
+    },
+    [dispatch],
+  );
 
-    if (addAccount.fulfilled.match(res)) {
-      emitMetric(METRIC_NAMES.accountScreenAddAccount, {
-        // eslint-disable-next-line @typescript-eslint/naming-convention
-        number_of_accounts: res.payload.allAccounts.length,
-      });
-      navigateTo(ROUTES.account);
-    }
+  const handleEnterPassword = async (values: FormValues) => {
+    await handleAddAccount(values.password);
   };
 
+  // If we have a private key we can assume the user password is also saved in
+  // the current session store, so no need to ask for it again
+  useEffect(() => {
+    if (hasPrivateKey) {
+      handleAddAccount();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
   useEffect(
     () => () => dispatch(clearApiError()) as unknown as void,
     [dispatch],
   );
 
+  // No need to ask for password if it's already stored, so let's just briefly
+  // display a loading screen while user is being redirected to next screen
+  if (hasPrivateKey) {
+    return (
+      <React.Fragment>
+        <SubviewHeader title="" />
+        {!authError && <Loading />}
+        {authError && <AppError>{authError}</AppError>}
```

Comment:
What are the scenarios where this might happen? I think generally it'd be great to give the user a way to fix whatever went wrong. For ex, if we think that this was due to us somehow having an incorrect password in the `background`, maybe we should show them the `Enter Password` field so they can put the correct password in

Reply:
@piyalbasu Tbh I'm not sure yet about all the scenarios which could result in an auth error here. I was basically trying to guard against this error since [we were already doing it](https://github.com/stellar/freighter/blob/master/extension/src/popup/views/AddAccount/AddAccount/index.tsx#L37). But I see what you mean, in the old UX the user would always have the chance to input the password again. 🤔 

I think I'll make a change so that if there is an auth error the user will be redirected to the "Enter Password" screen with the error message right below the input inbox so we keep the same old UX in this case allowing users to input the password again. What do you think?

Reply:
@piyalbasu on this commit I've changed the error UX to be like this: